### PR TITLE
Filter overrepresented barcodes

### PR DESCRIPTION
**Pull request by acorbat**
My strategy for #40 was to add the option to read a CSV where each row is a CloneID to be ignored. The is_similar function will be used to compare CloneIDs detected with the ones in the blacklist (with a maximum Hamming distance of 0 in order to also remove the ones missing bases).

### CODE REVIEW

File: src/trex/cli/run10x.py [hunk 1]
```diff
@@ -334,6 +356,18 @@ def read_allowed_cellids(path):
     return set(allowed_ids)
 
 
+def read_clone_id_blacklist(path: Path) -> List:
+    """
+    Read a user-provided list of CloneIDs to be ignored from a CSV
```
[the remark this answers is not in the document]
Comment (acorbat) [reply src/trex/cli/run10x.py:361]: I haven't gone through smartseqs and I am not sure what the differences are to make this adaptation. For now, I would leave it here and open an issue to adapt it for other pipes.

File: README.md [hunk 2]
```diff
@@ -200,6 +200,16 @@ Example:
 1	CACTCGTGGTACCACAAGCA
 ```
 
+## Filter cloneID
+
+Text file of one cloneIDs per line to ignore during the TREX run.
```
Comment (marcelm) [inline README.md:205]:
```suggestion
Text file with cloneIDs to ignore. The format is one cloneID per line.
```

File: README.md [hunk 3]
```diff
@@ -200,6 +200,16 @@ Example:
...
+Adding this file via the `--filter-cloneids` option allows to ignore cloneIDs that have been identified as overrepresented or artefactual during library characterization.
+Example file
```
Comment (marcelm) [inline README.md:207]:
```suggestion
Example file:
```

File: src/trex/cli/__init__.py [hunk 4]
```diff
@@ -159,6 +159,12 @@ def add_common_arguments(parser, smartseq: bool):
         "Expected format: see documentation",
         default=None,
     )
+    filter_group.add_argument(
+        "--filter-cloneids",
+        type=Path,
+        help="Text file containing one cloneID per line that are to be ignored during the analysis. "
+        "This flag enables to remove cloneIDs e.g. overrepresented cloneIDs, misalignments.",
```
Comment (marcelm) [inline src/trex/cli/__init__.py:166]:
```suggestion
        help="Text file with cloneIDs to be ignored during the analysis. Format: One cloneID per line. "
        "Use this to remove e.g. overrepresented cloneIDs or misalignments.",
```

File: CHANGES.md [hunk 5]
```diff
@@ -9,3 +9,4 @@
   subclusters are detected and removed. The cell IDs of detected doublets
   is written to `doublets.txt`. Doublet detection can be disabled with
   `--keep-doublets`.
+* [#40](https://github.com/frisen-lab/TREX/issues/40): Added `--filtering-cloneids` option for excluding certain cloneIDs
```
Comment (marcelm) [inline CHANGES.md:12]:
```suggestion
* [#40](https://github.com/frisen-lab/TREX/issues/40):
  Added `--filter-cloneids` option for excluding certain cloneIDs
```

File: README.md [hunk 6]
```diff
@@ -200,6 +200,16 @@ Example:
 1	CACTCGTGGTACCACAAGCA
 ```
 
+## Filter cloneID
```
Comment (marcelm) [inline README.md:203]: My comment about "filtering" (with "ing") referred to the title of this section.

```suggestion
## Filtering cloneIDs
```

**Pull request merged**